src/plugins/intel_gpu/src/runtime/ocl/ocl_device.cpp

```diff
@@ -119,12 +119,20 @@ device_type get_device_type(const cl::Device& device) {
     return unified_mem ? device_type::integrated_gpu : device_type::discrete_gpu;
 }
 
-gfx_version parse_version(cl_uint ver) {
-    uint16_t major = ver >> 16;
-    uint8_t minor = (ver >> 8) & 0xFF;
-    uint8_t revision = ver & 0xFF;
+gfx_version parse_version(cl_uint gmdid) {
+    union GMDID {
+        uint32_t value;
+        struct
+        {
+            uint32_t revision : 6;
+            uint32_t reserved : 8;
+            uint32_t release : 8;
+            uint32_t architecture : 10;
+        };
+    };
 
-    return {major, minor, revision};
+    GMDID gmd_id = {gmdid};
```

Comment:
I assume that such parsing will not work for older driver versions. Ideally we need a backward compatible logic here. What if we try to parse id with a new format, then check if the results is meaningful (e.g. check that arch in a range [0, 30] or it's one of [9,10,11,12] list - something like that), and fallback to the previous logic otherwise.